### PR TITLE
Add autocorrection to the syntactic sugar rule

### PR DESCRIPTION
**Pull request by vani2**
I've added autocorrection to the rule:

```swift
let x: Array<String> /* will be */ let x: [String]
let x: Dictionary<Int, String> /* will be */ let x: [Int: String]
let x: Optional<Int> /* will be */ let x: Int?
let x: ImplicitlyUnwrappedOptional<Int> /* will be */ let x: Int!
func x(a: Array<Int>, b: Int) -> [Int: Any] /* will be */ func x(a: [Int], b: Int) -> [Int: Any]
func x(a: [Int], b: Int) -> ↓Dictionary<Int, String> /* will be */ func x(a: [Int], b: Int) -> [Int: String]
let x = Array<String>.array(of: object) /* will be */ let x = [String].array(of: object)
let x: Swift.Optional<String> /* will be */ let x: String?
```

### CODE REVIEW

**Comment by marcelofabri** [comment]
Can you run the unit tests locally to update the docs? Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,8 @@
   using `flatMap` over `map { ... }.reduce([], +)`.  
   [Marcelo Fabri](https://github.com/marcelofabri)
   [#2883](https://github.com/realm/SwiftLint/issues/2883)
+* Add autocorrection to `syntactic_sugar`.
```
Comment (marcelofabri) [inline CHANGELOG.md]: please add an empty line before this entry

**Comment by vani2** [comment]
> Can you run the unit tests locally to update the docs? Thanks!

I did that, that has no effect. So I've updated `Rules.md` manually.

**Comment by marcelofabri** [comment]
@vani2 I think this was failing because of some other issues in CI that are fixed in `master`. I rebased and force pushed to your branch, hope that's ok.

**Comment by marcelofabri** [comment]
Thanks @vani2 🎉

**Pull request merged**